src/hooks/useProjectVariablesUpdate/index.ts

```diff
@@ -71,7 +103,65 @@ export const useProjectVariablesUpdate = (
         return;
       }
 
-      operations = [...operations, ...makeVariableValueOperations(variable, pendingValues, [])];
+      operations = [
+        ...operations,
+        ...makeVariableValueOperations({
+          pendingCellValues: pendingValues,
+          pendingValues: [],
+          variable,
+        }),
+      ];
+    });
+
+    // handle image updates
+    pendingImages.forEach((pendingValues) => {
+      pendingValues.forEach((image) => {
+        const newValue = { type: image.type, citation: image.citation, caption: image.caption };
+        operations.push({
+          operation: 'Update',
+          valueId: image.id,
+          value: newValue,
+          existingValueId: image.id,
+        });
+      });
+    });
+
+    // handle image deletions
+    pendingDeletedImages.forEach((pendingValues) => {
+      pendingValues.forEach((deletedImage) => {
+        operations.push({
+          operation: 'Delete',
+          valueId: deletedImage.id,
+          existingValueId: deletedImage.id,
+        });
+      });
+    });
+
+    // handle image uploads
+    pendingNewImages.forEach((pendingValues, variableId) => {
+      const variable = variablesWithValues.find((variableWithValues) => variableWithValues.id === variableId);
+      if (!variable) {
+        // This is impossible if the form is only displaying variables that were initialized within the hook
+        snackbar.toastError(strings.GENERIC_ERROR);
+        return;
+      }
+
+      pendingValues.forEach((newImage, index) => {
+        const upRequest = dispatch(
+          requestUploadImageValue({
+            variableId: variable.id,
+            file: newImage.file,
+            caption: newImage.caption,
+            citation: newImage.citation,
+            projectId,
+          })
+        );
+
+        // set request id with last image request
+        if (pendingValues.length - 1 === index) {
+          setUploadRequestId(upRequest.requestId);
```

Comment:
This is only tracking the progress of one image upload for now. Do we have a method for tracking the progress of multiple dispatched requests?

Reply:
No. What we have in other parts of the app, is a thunk that receives all the request, and the response is only one, after all are completed: https://github.com/terraware/terraware-web/blob/main/src/redux/features/participantProjectSpecies/participantProjectSpeciesAsyncThunks.ts#L49

Reply:
Ok I'll create a new thunk using that as a reference, thanks @constanzauanini! 🙌